core/model/src/main/java/org/eclipse/rdf4j/model/util/Statements.java

```diff
@@ -26,6 +31,16 @@
  */
 public class Statements {
 
+	/**
+	 * A {@link Function} that maps {@link Triple} to {@link org.eclipse.rdf4j.model.BNode} consistently. Multiple
+	 * invocations for the same {@link Triple} will return the same {@link org.eclipse.rdf4j.model.BNode}.
+	 *
+	 * The current implementation creates a {@link org.eclipse.rdf4j.model.BNode} by encoding the string representation
+	 * of the {@link Triple} using base64 URL-safe encoding.
+	 */
+	public static Function<Triple, Resource> TRIPLE_BNODE_MAPPER = (t) -> SimpleValueFactory.getInstance()
+			.createBNode(Base64.getUrlEncoder().encodeToString(t.stringValue().getBytes(StandardCharsets.UTF_8)));
+
```

Comment:
Does this risk bnode identifiers being really long of the Triple contains a long literal?

Reply:
I don't see that as a major issue, but if it becomes a problem we could switch to using a fixed-length message digest hash (MD5) as an alternative. I'd suggest it's YAGNI for now though.

Reply:
Would collisions be an issue then?

Reply:
IIRC collision risk of MD5 is 1:2^128. For our purposes safe enough I'd say.